test/hts-precompile/utils.js

```diff
@@ -668,6 +671,106 @@ class Utils {
     const txResponse = await signTx.execute(signerClient)
     await txResponse.getReceipt(signerClient)
   }
+
+  /**
+   * @dev Prepares an array of typed IHederaTokenService.TokenKey for HTS token creation transactions.
+   *
+   * @notice IHederaTokenService.KeyValue supports a range of systems (e.g. ED25519, ECDSA secp256k1, etc.).
+   *         Refer to the documentation at https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L116
+   *         for more information on supported key types.
+   *
+   * @notice This function sets the `contractEVMAddress` (the address of the deployed HTS example contract) as the key for all key types (e.g. ADMIN, KYC, SUPPLY, etc.) for testing purposes.
+   *         If any other `keys` besides the `contractEVMAddress` are desired to be set, the keys must first sign an AccountUpdate transaction.
+   *         Refer to `this.updateAccountKeysViaHapi()` for usage examples on signing with `keys`.
+   */
+  static prepareTokenKey(contractEVMAddress, ECDSAPubKey) {
+    return [
+      {
+        keyType: 1, // 0th bit: ADMIN
+        key: {
```

Comment:
This is conflicting logic.
The key type should be of one kind but you're setting all the value.
If the key is a contractId then set contract id, if it's ed25519 then set that, if it's an ecdsa then set that etc.